test/tests/meshgenerators/file_mesh_generator/2d_diffusion_iga.i

```diff
@@ -0,0 +1,52 @@
+[Mesh]
+  [cyl2d_iga]
+    type = FileMeshGenerator
+    file = PressurizedCyl_Patch6_4Elem.e
+  []
+  allow_renumbering = false
+  parallel_type = replicated
+[]
+
+[Variables]
+  [u]
+    order = SECOND  # Must match mesh order
```

Comment:
See https://github.com/idaholab/moose/pull/20646#issuecomment-1084979056 - the answer is "no", which is awful, but also "that's already a problem with non-IGA runs" and "there's no good way to fix that at the Moose level".  I'm hoping you'll agree that puts it out of the scope of this PR?

Reply:
Is it possible to get the mesh order after the entire mesh is already loaded? If so, we should be able to compare the FEM order with the mesh order after the mesh being loaded.

In the current state, I do feel a little sad on this. I am pretty sure dump Fande will spend several hours on debugging if he specifies a wrong FEM order for a given mesh before realizing that what he should do

Reply:
There's no such thing as "the mesh order" - users are mixing different elem orders in the same mesh in some codes.  Plus, it'd be a full memory sweep's extra expense, and worse still it'd give us false positives; "load a mesh then run it through `all_second_order()`" is a very useful workflow.  libMesh gives users loads of flexibility to change things right up until the last minute, which unfortunately makes it hard to be sure they aren't going to fix an apparent inconsistency right before the last minute.

I think the right fix then is "test at the very last minute" - I'll work up something that inspects elements as we're assigning DoF indices, so we can throw errors with more informative messages than the low-level errors we otherwise hit.  But again, that's a method in libMesh, not Moose.  And even if it was a fix to go in at the Moose level, would this be the right place to put it?  Surely it'd be a bit much to block a new feature addition unless it simultaneously fixes a 10-year-old UI inadequacy?

Reply:
I am fine if we want to do this in a follow-up PR